p2panda-auth/src/group/mod.rs

```diff
@@ -534,66 +546,66 @@ where
         // TODO: this is a bit of a sanity check, if we want to check for duplicate operation
         // processing here in the groups api then there should probably be a hashset of operations
         // ids maintained on the struct for efficient lookup.
-        if y.operations.iter().any(|op| op.id() == operation_id) {
+        if y.operations
+            .iter()
+            .find(|op| op.id() == operation_id)
+            .is_some()
+        {
+            // The operation has already been processed.
             return Err(GroupError::DuplicateOperation(operation_id, group_id));
         }
 
         if y.group_id != group_id {
-            // This operation is not intended for this group.
+            // The operation is not intended for this group.
             return Err(GroupError::IncorrectGroupId(group_id, y.group_id));
         }
 
-        // The resolver implementation contains the logic which determines when rebuilds are
-        // required.
-        //
-        // TODO: before performing this check we want to actually apply the operation to the
-        // group. This will allow us to handle any validation which occur at that point already.
-        if RS::rebuild_required(&y, operation) {
-            // Add all new operations to the graph and operations vec.
-            y.graph.add_node(operation.id());
-            for previous in previous_operations {
-                y.graph.add_edge(previous, operation.id(), ());
-            }
-            y.operations.push(operation.clone());
-
-            // Perform the re-build and return the new state.
-            return Self::rebuild(y);
-        }
-
-        // Compute the members state by applying the new operation to it's claimed "previous"
+        // Compute the member's state by applying the new operation to it's claimed "previous"
         // state.
         //
         // This method validates that the actor has permission to perform the action.
-        match control_message {
-            GroupControlMessage::GroupAction { action, .. } => {
-                y = Self::apply_action(
-                    y,
-                    operation_id,
-                    GroupMember::Individual(actor),
-                    &previous_operations,
-                    &action,
-                )?;
-            }
-            // No action required as revokes would have triggered a rebuild in the previous step.
-            //
+        //
+        // Do not update the state if the attempt to apply the action fails.
+        let mut y_i = match control_message {
+            // TODO: either here or inside apply_action we need to account for expected errors
```

Comment:
I believe this is the cause of one of the test failures, as once some actions get filtered out, then later ones may trigger an error (eg. applying a now "ignored" member's action)